yetanotherco/aligned_layer
# feat(ffi): ffi refactors

Refactors ffi bindings after feedback from @Oppen about #904:
- Removes redundant length parameters in go code.
- Refactors how a zero length public input buffer is handled in risc0-vm.
- Cleans up tests.
- Adds zero length check to `listRefToBytes()`

## To Test:
- Run all FFI tests and verify they pass:
- `make test_halo2_ipa_go_bindings_macos`
- `make test_halo2_kzg_go_bindings_macos`
-  `make test_risc_zero_go_bindings_macos`
- `make test_sp1_go_bindings_macos`
- `make test_merkle_tree_go_bindings_macos`
- `make test_merkle_tree_old_go_bindings_macos`
